scripts/leapp_upgrade.py

```diff
@@ -0,0 +1,382 @@
+    (print_cmd=True) and environment variables in form of dictionary(env=None).
+    Switching off printing the command can be useful in case it contains a
+    password in plain text.
+
+    The cmd is specified as a list starting with the command and followed by a
+    list of arguments. Example: ["yum", "install", "<package>"]
+    """
+    if isinstance(cmd, str):
+        raise TypeError("cmd should be a list, not a str")
+
+    if print_cmd:
+        print("Calling command '%s'" % " ".join(cmd))
+
+    process = subprocess.Popen(
+        cmd, stdout=subprocess.PIPE, stderr=subprocess.STDOUT, bufsize=1, env=env
+    )
+
+    output = ""
+
+    if not wait:
+        return output, None
+
+    for line in iter(process.stdout.readline, b""):
+        line = line.decode("utf8")
+        output += line
+
+    # Call wait() to wait for the process to terminate so that we can
+    # get the return code.
+    process.wait()
+
+    return output, process.returncode
+
+
+def check_if_package_installed(pkg_name):
+    _, return_code = run_subprocess(["rpm", "-q", pkg_name])
+    return return_code == 0
+
+
+def setup_leapp(command, rhui_pkgs):
+    print("Installing leapp ...")
+    output, returncode = run_subprocess(command)
+    if returncode:
+        print(
+            "Installation of leapp failed with code '%s' and output: %s\n"
+            % (returncode, output)
+        )
+        raise ProcessError(
+            message="Installation of leapp failed with code '%s'." % returncode
+        )
+
+    print("Check installed rhui packages ...")
+    for pkg in rhui_pkgs:
+        if check_if_package_installed(pkg["src_pkg"]):
+            pkg["installed"] = True
+    return [pkg for pkg in rhui_pkgs if pkg.get("installed", False)]
+
+
+def should_use_no_rhsm_check(rhui_installed, command):
+    print("Checking if subscription manager and repositories are available ...")
+    rhsm_repo_check_fail = True
+    _, rhsm_installed_check = run_subprocess(["which", "subscription-manager"])
+    if rhsm_installed_check == 0:
+        rhsm_repo_check, _ = run_subprocess(
+            ["subscription-manager", "repos", "--list-enabled"]
+        )
+        rhsm_repo_check_fail = (
+            "This system has no repositories available through subscriptions."
+            in rhsm_repo_check
+            or "Repositories disabled by configuration." in rhsm_repo_check
+        )
+
+    if rhui_installed and not rhsm_repo_check_fail:
+        print("RHUI packages detected, adding --no-rhsm flag to preupgrade command")
+        command.append("--no-rhsm")
+        return True
+    return False
+
+
+def install_leapp_pkg_corresponding_to_installed_rhui(rhui_pkgs):
+    print("Installing leapp package corresponding to installed rhui packages")
+    for pkg in rhui_pkgs:
+        install_pkg = pkg["leapp_pkg"]
+        install_output, returncode = run_subprocess(
+            ["yum", "install", "-y", install_pkg]
```

Comment:
Full path to yum as well. This is something that we need to change in oamg/convert2rhel-worker-scripts too.

Reply:
Did it for convert2rhel in https://github.com/oamg/convert2rhel-worker-scripts/pull/29/commits/e2dc78f5984a3dfb12a0e76f66d7991727523716 , for preupgrade in already mentioned https://github.com/oamg/leapp-insights-tasks/pull/4